pandas/tests/tools/test_to_datetime.py

```diff
@@ -125,24 +126,21 @@ def test_to_datetime_format_YYYYMMDD_with_nat(self, cache):
         expected[2] = np.nan
```

Comment:
Don't need to have `expected` if it raises.  Although the reason it is raising is because `np.nan` changes the `dtype` to float.  So this could be a loss of functionality when people have missing data.

Reply:
thanks - I've marked as draft as there might be a better solution out there which doesn't degrade performance, just working on it